packages/browser/src/transports/fetch.ts

```diff
@@ -1,13 +1,82 @@
 import { eventToSentryRequest, sessionToSentryRequest } from '@sentry/core';
-import { Event, Response, SentryRequest, Session } from '@sentry/types';
-import { getGlobalObject, supportsReferrerPolicy, SyncPromise } from '@sentry/utils';
+import { Event, Response, SentryRequest, Session, TransportOptions } from '@sentry/types';
+import { getGlobalObject, logger, supportsReferrerPolicy, SyncPromise } from '@sentry/utils';
 
 import { BaseTransport } from './base';
 
-const global = getGlobalObject<Window>();
+type FetchImpl = typeof fetch;
+
+/**
+ * A special usecase for incorrectly wrapped Fetch APIs in conjunction with ad-blockers.
+ * Whenever someone wraps the Fetch API and returns the wrong promise chain,
+ * this chain becomes orphaned and there is no possible way to capture it's rejections
+ * other than allowing it bubble up to this very handler. eg.
+ *
+ * const f = window.fetch;
+ * window.fetch = function () {
+ *   const p = f.apply(this, arguments);
+ *
+ *   p.then(function() {
+ *     console.log('hi.');
+ *   });
+ *
+ *   return p;
+ * }
+ *
+ * `p.then(function () { ... })` is producing a completely separate promise chain,
+ * however, what's returned is `p` - the result of original `fetch` call.
+ *
+ * This mean, that whenever we use the Fetch API to send our own requests, _and_
+ * some ad-blocker blocks it, this orphaned chain will _always_ reject,
+ * effectively causing another event to be captured.
+ * This makes a whole process become an infinite loop, which we need to somehow
+ * deal with, and break it in one way or another.
+ *
+ * To deal with this issue, we are making sure that we _always_ use the real
+ * browser Fetch API, instead of relying on what `window.fetch` exposes.
+ * The only downside to this would be missing our own requests as breadcrumbs,
+ * but because we are already not doing this, it should be just fine.
+ *
+ * Possible failed fetch error messages per-browser:
+ *
+ * Chrome:  Failed to fetch
+ * Edge:    Failed to Fetch
+ * Firefox: NetworkError when attempting to fetch resource
+ * Safari:  resource blocked by content blocker
+ */
+function getNativeFetchImplementation(): FetchImpl {
+  // Make sure that the fetch we use is always the native one.
+  const global = getGlobalObject<Window>();
+  const document = global.document;
+  // eslint-disable-next-line deprecation/deprecation
+  if (typeof document?.createElement === `function`) {
+    try {
+      const sandbox = document.createElement('iframe');
+      sandbox.hidden = true;
+      document.head.appendChild(sandbox);
+      if (sandbox.contentWindow?.fetch) {
+        return sandbox.contentWindow.fetch.bind(global);
+      }
+      document.head.removeChild(sandbox);
```

Comment:
This is so slow, that it doesn't execute below 50ms for mobile lighthouse. Thus it delays the TTI metric in lighthouse, even if you only execute install in a single macrotask, thus including sentry now costs you points in mobile lighthouse and lowers your ranking on google.

# desktop
![image](https://user-images.githubusercontent.com/905328/111813333-5927e680-88d9-11eb-8658-7fc3279cbb0e.png)

# mobile
![image](https://user-images.githubusercontent.com/905328/111813770-ecf9b280-88d9-11eb-926b-416bb8a9d026.png)

Reply:
This was 6.0.1 btw., so quite some regression

![image](https://user-images.githubusercontent.com/905328/111821911-7c579380-88e3-11eb-987b-10b492f16429.png)

Reply:
Thanks for letting us know, should be patched in https://github.com/getsentry/sentry-javascript/pull/3341